insights/reducers/grub_conf.py

```diff
@@ -71,55 +67,54 @@
 class GrubConf(object):
     """Process Grub configuration v1 or v2 based on which type is passed in
 
     Attributes:
-        kernel_initrds (list)
-        is_kdump_iommu_enabled (bool)
+        version (int): returns 1 or 2, version of the GRUB configuration
+        is_efi (bool): returns true if grub configuration for EFI-based system
+        kernel_initrds (dict): returns a dict of the `kernel` and `initrd`
+            files referenced in GRUB configuration files
+        is_kdump_iommu_enabled (bool): returns true if any kernel entry
+            contains "intel_iommu=on"
 
     Raises:
-        Exeption: If shared is not Grub1Config pr Grub2Config
+        Exception: If shared is not Grub1Config, Grub2Config or Grub2EFIConfig
+
     """
 
     def __init__(self, local, shared):
 
-        self.kernel_initrds = []
-        self.crash_kernel_offset = ""
-        self.is_kdump_iommu_enabled = False
-
-        # get grub configuartion
-        self.grub1 = shared.get(Grub1Config)
-        self.grub2 = shared.get(Grub2Config)
-
-        # grub1
-        if (self.grub1):
-            return self._get_grub1_info()
-
-        # grub2
-        elif (self.grub2):
-            return self._get_grub2_info()
-
-        raise Exception("Not Grub v1 or Grub v2.")
-
-    def _get_grub1_info(self):
-        """
-        Process Grub v1 configuration file.
+        # get grub configuration
+        self.grub = None
+        if Grub1Config in shared:
+            self.grub = shared.get(Grub1Config)
+        elif Grub2Config in shared:
+            self.grub = shared.get(Grub2Config)
+        elif Grub2EFIConfig in shared:
+            self.grub = shared.get(Grub2EFIConfig)
+        else:
+            raise Exception("No Grub Configuration.")
+
+        self.version = self.grub._version
+        self.is_efi = self.grub._efi
+        self.is_kdump_iommu_enabled = self.grub.is_kdump_iommu_enabled
+        self.kernel_initrds = self.grub.kernel_initrds
+
+    def get_grub_cmdlines(self, search_text):
         """
+        Get the boot entries in which `cmdline` contains the `search_text`
 
-        self.is_kdump_iommu_enabled = self.grub1.is_kdump_iommu_enabled
-        self.kernel_initrds = self.grub1.kernels_initrds
-
-    def _get_grub2_info(self):
-
-        """
-        Process Grub v2 configuration file.
+        Returns:
+            list: a list of dictionaries for each boot entry which `cmdline`
+            contains the `search_text` in below format:
+            - 'name': "Red Hat Enterprise Linux Server"
+            - 'cmdline': "kernel /vmlinuz-2.6.32-431.11.2.el6.x86_64 crashkernel=128M rhgb quiet"
         """
-
-        self.is_kdump_iommu_enabled = False
-        self.kernel_initrds = self.grub2.kernels_initrds
+        if not search_text:
+            raise Exception("None or empty `search_text` is not allowed.")
+        return [entry for entry in self.grub.boot_entries if search_text in entry.cmdline]
```

Comment:
@xiangce would it be better to just return an empty list here rather than raise a generic exception?  If `search_text` is the wrong time than a different exception will be raised, and if `search_text` is `''` then all lines will be returned which doesn't seem to be the correct answer.